dhall/src/Dhall.hs

```diff
@@ -1702,6 +1700,49 @@ instance Inject a => Inject (Data.Set.Set a) where
 
 instance (Inject a, Inject b) => Inject (a, b)
 
+{-|
+
+>>> prettyExpr $ embed inject (Data.Map.fromList [(1 :: Integer, True )])
+[ { mapKey = +1, mapValue = True } ]
+
+>>> prettyExpr $ embed inject (Data.Map.fromList [] :: Data.Map.Map Integer Bool)
+[] : List { mapKey : Integer, mapValue : Bool }
+
+In case of duplicate keys, the last input is kept.
```

Comment:
`Map`s can't contain duplicate keys. What `Data.Map.fromList` does, needn't be documented here.

Reply:
I should not code after 11PM.

Reply:
No worries! :) We have code review and a good type checker. :)